File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        transform(s.begin(),s.end(),s.begin(),[](unsigned char c){
            return tolower(c);
        });
        set<char> mySet = {'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z','0','1','2','3','4','5','6','7','8','9'};
        string newS;
        int l = s.size();
        for(int i = 0;i<l;i++){
            if(mySet.count(s.at(i))>0){
                newS += s.at(i);
            }
        }
        int pL = 0;
        int pR = newS.size()-1;
        while(pL<pR){
            if(newS.at(pL)!=newS.at(pR)){
                return false;
            }
            pR--;
            pL++;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String newS = "";
        for(int i = 0;i<s.length();i++){
            char temp = s.charAt(i);
            int ye = temp;
            if(ye>=65 && ye<=90){
                ye = ye+32;
                char temp1 = (char) ye;
                newS = newS+temp1;
            }else if(ye>=97 && ye<=122){
                newS = newS+s.charAt(i);
            }else if(Character.isDigit(temp)){
                newS = newS+temp;
            }
        }
        //System.out.println(newS);
        if(newS.length()==0){
            return true;
        }
        if(newS.length()==1){
            return true;
        }
        
            int counter = newS.length()-1;
            for(int i = 0;i<newS.length()/2;i++){
                if(newS.charAt(i)!=newS.charAt(counter)){
                    return false;
                }
                counter--;
            }
        
        return true;
    }
}